### ISSUE TITLE
GetClustCfAll- Changes made in driver code to solve the error

### ISSUE THREAD

**Issue by murli1201** [NSSAC/CINES#87]
@cckuhlman @Lucaslhm

Hi Chris, Lucas,

For method "GetClustCfAll", on generating the output files using the driver code, the output files were created but we were getting an error as " Error in writing output for variable -- DegToCCfV -- of method ---- snap.GetClustCfAll(inputFile_Graph,graphType,srcCol,desCol,SampleNodes,outputFile_DegToCCfV,outputFile_libraryMethodReturnTypeVariable) ---- to file.")

As discussed in the meeting, I did the below changes according to the example in snap documentation, which resolved the error and correct output was generated-

(Original)
 for item in DegToCCfV:
 fhout.write(str(item.GetVal1()) + "," + str(item.GetVal2()) + "," + str(DegToCCfV[item]) + "\n")

(Edited)
 for item in DegToCCfV:
 fhout.write(str(item.GetVal1()) + "," + str(item.GetVal2()) + "\n")

Note: As we have made changes to the driver code manually, it may give error if the driver code is regenerated in future.

Is there any other solution?

Thank you,
Murli

**Comment by cckuhlman** [comment]
@murli1201 

This is similar to past issues about datatypes.

The method is here:  https://snap.stanford.edu/snappy/doc/reference/GetClustCfAll.html?highlight=getclustcfall

It says that DegToCCf is of type " vector of pairs of floats".  So this type, in this file, is just a text description.  It is not the actual data type that we seek.

So, go here, to the page of datatypes for snap:  https://snap.stanford.edu/snappy/doc/tutorial/tutorial.html?highlight=graph%20types

Now, search for "vector of pairs of floats".  There is, unfortunately, no match.   So, find this section:  SNAP Types in Snap.py on that page.
You look through the types and there is a type that says (about the 6th entry down):
TFltPrV, a vector of float pairs;

So, for this run, and to handle DegToCCf, you have to put this type---TFltPrV---into the snap text input file instead of "vector of pairs of floats".
This change to the input file--like all changes to the snap input file--should be noted in this file:  https://github.com/NSSAC/cines-snap-texts/blob/master/methods-to-implement/library-methods-to-implement.md
Then, in datatypes.dat file, you must have an entry for this datatype.  If there is one already for TFltPrV, then see whether the output template looks like what you want.  Per you above, it is:

(Edited)
for item in DegToCCfV:
fhout.write(str(item.GetVal1()) + "," + str(item.GetVal2()) + "\n")

... or is something close to this that will work.

In summary, to get what you want, this code snippet must be in a templated.  The way you get a template for a particular data type, is to enter the data type into the datatype.dat file, and to enter the names of templates for that data type, and then to make the templates.

chris

**Comment by murli1201** [comment]
Yes, i made the changes same as you said in the above comment today and everything worked correctly. I forgot to add comment actually.
Thanks.

**Comment by cckuhlman** [comment]
Can we close this issue?

**Comment by murli1201** [comment]
Yes, sure.
